tests/test_pytypes.py

```diff
@@ -220,6 +220,27 @@ def test_constructors():
         assert noconv2[k] is expected[k]
 
 
+def test_pybind11_str_raw_str():
+    # specifically to exercise pybind11::str::raw_str
+    cvt = m.convert_to_pybind11_str
+    assert cvt(u"Str") == u"Str"
```

Comment:
This could also be:

```python
pybind11_str = unicode if sys.version_info.major < 3 else str
for test in [u"Str",
             None,
             False,
             True,
             ...]:
    assert cvt(test) == pybind11_str(test)
```

Or do we need to be explicit here, line by line?

Reply:
I actually prefer it being explicit, just to make it as clear as possible, and permit specific branching.

Reply:
Also fine. Let's leave it up to @rwgk, then :-)